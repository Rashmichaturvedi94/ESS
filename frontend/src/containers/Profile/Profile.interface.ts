export interface ProfileProps {
  myProp?: boolean;
  /* Add props here. */
}
