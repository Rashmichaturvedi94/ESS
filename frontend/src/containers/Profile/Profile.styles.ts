import { Box } from '@material-ui/core';
import styled from 'styled-components';

export const Profile = styled.div`
  flex: 1;
  display: flex;
  flex-direction: row;
`;

export const IconBox = styled(Box)`
height: 200px;
width: 200px;
border-radius: 100px;
border-color: black;
border-width: 2px;
border-style: solid;
margin-right: 50px;
`;