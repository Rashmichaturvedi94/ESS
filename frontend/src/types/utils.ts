export type Nullable<T> = {
  [P in keyof T]: T[P] | null;
};

export type Uri = string;

export type DateString = string;

export type LocalOrRemoteFile = File | Uri;

export type Modify<T, R> = Omit<T, keyof R> & R;
