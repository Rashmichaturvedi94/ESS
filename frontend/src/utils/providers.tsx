import React, { ComponentType, FC } from 'react';
import * as Sentry from '@sentry/react';
import { QueryClientProvider } from 'react-query';
import DateFnsUtils from '@date-io/date-fns';
import { ThemeProvider } from 'styled-components';
import { StylesProvider, MuiThemeProvider } from '@material-ui/core';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import { queryClient } from '../api';
import { theme } from '../styles/theme';


const createProviderHOC = (Provider: ComponentType) => <P extends object>(
  Component: ComponentType<P>,
) => (props: P) => (
  <Provider>
    <Component {...props} />
  </Provider>
);
const FallbackComponent = () => <div>An error has occurred</div>;

export const Providers: FC = ({ children }) => (
  <Sentry.ErrorBoundary fallback={FallbackComponent} showDialog>
    <StylesProvider injectFirst>
      <MuiThemeProvider theme={theme}>
        <ThemeProvider theme={theme}>
          <MuiPickersUtilsProvider utils={DateFnsUtils}>
            <QueryClientProvider client={queryClient}>
              {children}
            </QueryClientProvider>
          </MuiPickersUtilsProvider>
        </ThemeProvider>
      </MuiThemeProvider>
    </StylesProvider>
  </Sentry.ErrorBoundary>
);

export const withProviders = createProviderHOC(Providers);
