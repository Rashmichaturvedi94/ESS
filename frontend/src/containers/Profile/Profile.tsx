import React, { FC } from 'react';
import { Box } from '@material-ui/core';
import PersonOutlineIcon from '@material-ui/icons/PersonOutline';
import { Profile as ProfileComponent, IconBox } from './Profile.styles';
import { ProfileProps } from './Profile.interface';
import { ProfileForm } from '../../components/ProfileForm';
import { useUserMe, useUpdateUserEmail } from '../../api';

export const Profile: FC<ProfileProps> = () => {
  const user  = useUserMe();
  const { mutate } = useUpdateUserEmail();

  return (
    <ProfileComponent>
      <Box flex={1} display="flex" alignItems="center" justifyContent="center">
        <IconBox>
          <PersonOutlineIcon style={{ color: 'black', fontSize: 200}}/>
        </IconBox>
        <Box display="flex" flexDirection="column" width={320} gridRowGap={20}>
          <ProfileForm
            initialValues={{name: user.data?.username ?? '', username: user.data?.username ?? '', email: user.data?.email ?? '', userId: user.data?.id ?? 0 }}
            onSubmit={(data) =>{
                mutate({ data, params: { userId: user.data?.id } });
            }
            }
          />
        </Box>
      </Box>
    </ProfileComponent>
  );
};
