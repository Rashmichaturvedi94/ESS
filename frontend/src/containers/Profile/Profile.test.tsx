import React from 'react';
import { render } from '../../utils/testUtils';
import { Profile } from './Profile';

describe('components/Profile', () => {
  it('should render', () => {
    const { container } = render(<Profile />);
    expect(container).toMatchSnapshot();
  });
});
