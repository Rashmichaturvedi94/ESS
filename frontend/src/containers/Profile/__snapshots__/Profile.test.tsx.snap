// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Profile should render 1`] = `
<div>
  <div
    class="sc-bdnxRM bOAqhc"
  >
    <div
      class="MuiBox-root MuiBox-root-1"
    >
      <div
        class="MuiBox-root MuiBox-root-2 sc-gtsrHT itGaod"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          style="color: black; font-size: 200px;"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
      <div
        class="MuiBox-root MuiBox-root-3"
      >
        <form>
          <div
            class="MuiBox-root MuiBox-root-4"
          >
            <div
              class="MuiFormControl-root MuiTextField-root sc-eCApnc cyyBMT MuiFormControl-fullWidth"
            >
              <div
                class="MuiInputBase-root MuiInput-root Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled"
                  disabled=""
                  name="username"
                  placeholder="Username"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root sc-hKFxyN GcSgM MuiFormControl-fullWidth"
            >
              <div
                class="MuiInputBase-root MuiInput-root MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  name="name"
                  placeholder="Name"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root sc-jSFjdj kssSvR MuiFormControl-fullWidth"
            >
              <div
                class="MuiInputBase-root MuiInput-root MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  name="email"
                  placeholder="Email"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Save
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
